tools/dump_tool/dump_tool.c

```diff
@@ -244,6 +244,50 @@ static int fsdump_recv(int dev_fd)
 	return 0;
 }
 
+static int extfsdump_recv(int dev_fd)
+{
+        int ret;
+        uint32_t ext_smartfs_size;
+        char ext_fs_bin[BINFILE_NAME_SIZE] = { '\0' };
+        FILE *fs_fp;
+
+        /* Receive Userfs flash partition size */
+        ret = b_read(dev_fd, (uint8_t *)&ext_smartfs_size, 4);
+        if (ret != 4) {
+                printf("Receiving Userfs size failed, ret = %d\n", ret);
+                return -1;
+        }
+
+	if (ext_smartfs_size == 0) {
+		printf("No external userfs partition found. Please check config\n");
+		return 0;
```

Comment:
Do you have any special reason to return 0 for this failure case? If not, I think `-1` is better.

Reply:
In case no External Userfs partition is there, I'd like the user to still have the option to continue with the dump tool and dump other partitions.
Hence, not returning -1 to avoid exiting the Dump Tool